go/consensus/cometbft/roothash/roothash.go

```diff
@@ -574,7 +580,12 @@ func (sc *serviceClient) handleTrackRuntime(ctx context.Context, height int64, r
 	}
 	sc.trackedRuntime[runtimeID] = tr
 	// Request subscription to events for this runtime.
-	sc.queryCh <- app.QueryForRuntime(tr.runtimeID)
+	// This has to be done only after history reindex is done, so that we don't
```

Comment:
This comment is not clear. This code needs to execute after reindex is done, but still executes if `GetRuntimeState` fails :thinking: